src/vs/workbench/contrib/chat/browser/chatParticipantContributions.ts

```diff
@@ -218,13 +218,16 @@ export class ChatExtensionPointHandler implements IWorkbenchContribution {
 						description: string;
 						examples: string[];
 					}[] = [];
-					if (providerDescriptor.disambiguation?.length) {
-						participantsAndCommandsDisambiguation.push(...providerDescriptor.disambiguation);
-					}
-					if (providerDescriptor.commands) {
-						for (const command of providerDescriptor.commands) {
-							if (command.disambiguation?.length) {
-								participantsAndCommandsDisambiguation.push(...command.disambiguation);
+
+					if (isProposedApiEnabled(extension.description, 'contribChatParticipantDetection')) {
+						if (providerDescriptor.disambiguation?.length) {
```

Comment:
You should remove `disambiguation` from the jsonSchema or say that it requires the proposal. And you could warn when someone included these but doesn't have the proposal. I get that you don't want to fail because someone might have published an extension already that uses them.

Reply:
Thanks!